### PR TITLE
Add support for changing ublox rate (c++) 

### PR DESCRIPTION
**Pull request by IgorAnohin**

### CODE REVIEW

File: C++/Navio/Common/Ublox.cpp [hunk 1]
```diff
@@ -346,6 +347,55 @@ int Ublox::testConnection()
...
+    msg.measure_rate = meas_rate;
+    msg.nav_rate     = nav_rate;
+    msg.timeref      = timeref;
+
+    _sendMessage(CLASS_CFG, MSG_CFG_RATE, &msg, sizeof(CfgNavRate));
+}
+
+int Ublox::_sendMessage(std::uint8_t msg_class, std::uint8_t msg_id, void *msg, std::uint16_t size)
+{
+    unsigned char buffer[buffer_length];
```
Comment (staroselskii) [inline C++/Navio/Common/Ublox.cpp]: it's a const. Make it `UBX_BUFFER_LENGTH`

File: C++/Navio/Common/Ublox.cpp [hunk 2]
```diff
@@ -346,6 +347,55 @@ int Ublox::testConnection()
...
+    int offset = _spliceMemory(buffer, &header, sizeof(UbxHeader));
+    offset = _spliceMemory(buffer, msg, size, offset);
+
+    auto checksum = _calculateCheckSum(buffer, offset);
+    offset = _spliceMemory(buffer, &checksum, sizeof(CheckSum), offset);
+
+    return SPIdev::transfer(spi_device_name.c_str(), buffer, nullptr, offset, 200000);
+}
+
+int Ublox::_spliceMemory(unsigned char *dest, const void *src, std::size_t size, int dest_offset)
```
Comment (staroselskii) [inline C++/Navio/Common/Ublox.cpp]: `src` should be `const`

File: C++/Navio/Common/Ublox.cpp [hunk 3]
```diff
@@ -346,6 +347,55 @@ int Ublox::testConnection()
...
+{
+    std::memmove(dest + dest_offset, src, size);
+    return dest_offset + size;
+}
+
+Ublox::CheckSum Ublox::_calculateCheckSum(unsigned char message[], std::size_t size) {
```
Comment (staroselskii) [inline C++/Navio/Common/Ublox.cpp]: `unsigned char *message` is more conventional

File: C++/Navio/Common/Ublox.h [hunk 4]
```diff
@@ -40,9 +40,13 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <cstdio>
 #include <iostream>
 #include <string>
+#include <stdint.h>
 #include <vector>
+#include <list>
```
Comment (staroselskii) [inline C++/Navio/Common/Ublox.h]: Looks like a left-over

**Review by staroselskii — CHANGES_REQUESTED** [review]
Almost fine. std::memmove stuff might actually be better inlined but this is not really something to be bothered.

File: C++/Navio/Common/Ublox.cpp (hunk 1 above)
Comment (IgorAnohin) [reply C++/Navio/Common/Ublox.cpp]: Done

File: C++/Navio/Common/Ublox.h (hunk 4 above)
Comment (IgorAnohin) [reply C++/Navio/Common/Ublox.h]: Yes, you're right.

File: C++/Navio/Common/Ublox.cpp (hunk 3 above)
Comment (IgorAnohin) [reply C++/Navio/Common/Ublox.cpp]: Fixed

**Pull request merged**